source/navitiacommon/navitiacommon/models/__init__.py

```diff
@@ -589,19 +589,13 @@ def delete_dataset(self, _type):
 
         if not result:
             return 0
-        job_list = {}
         for dataset, job in result:
             # Cascade Delete not working so delete Metric associated manually
             db.session.query(Metric).filter(Metric.dataset_id == dataset.id).delete()
             db.session.delete(dataset)
 
             # Delete a job without any dataset
-            if job.id not in job_list:
-                job_list[job.id] = 1
-            else:
-                job_list[job.id] += 1
-
-            if not (len(job.data_sets.all()) - job_list[job.id]):
+            if not len(job.data_sets.all()):
```

Comment:
So this reverts https://github.com/CanalTP/navitia/commit/4b86af2a020dc22559a6216030e273fe1fc3b98b
But I don't really understand what it did before.

Reply:
I doesn't revert it. it just does the same thing with one updated condition.
This function deletes datasets of a specific type. The goal of these lines is to delete a job with no more dataset associated.
-> Previoulsy, the request `job.data_sets.all()` wasn't updated wihtout the `db.commit()` so we had to calculate the remaining datasets associated. 
-> Now, the request `job.data_sets.all()` takes into the account the deletion so it already returns the remaining datasets associated.
This generated a warning only in Tyr tests that waited to be fixed! Here and now!